### PR TITLE
Change instanceGroup manager logic to use zone information from GCE when removing nodes

### PR DESCRIPTION
**Pull request by songrx1997**
/assign @swetharepakula

### CODE REVIEW

File: pkg/instancegroups/manager.go [hunk 1]
```diff
@@ -297,10 +298,27 @@ func (m *manager) add(groupName string, names []string) error {
 }
 
 // Remove removes the given instances from the appropriately zoned Instance Group.
-func (m *manager) remove(groupName string, names []string) error {
+func (m *manager) remove(groupName string, names []string, nameZoneMap map[string]string) error {
 	events.GlobalEventf(m.recorder, core.EventTypeNormal, events.RemoveNodes, "Removing %s from InstanceGroup %q", events.TruncatedStringList(names), groupName)
 	var errs []error
-	for zone, nodeNames := range m.splitNodesByZone(names) {
```
Comment (swetharepakula) [inline pkg/instancegroups/manager.go:303]: can we use a single loop to do everything?

You could still use the zone, nodeNames list and just add all the node names into the map as the first step?

Comment (swetharepakula) [reply pkg/instancegroups/manager.go:303]: misread the diff. This commend can be ignored.

File: pkg/instancegroups/manager.go [hunk 2]
```diff
@@ -187,27 +187,28 @@ func (m *manager) DeleteInstanceGroup(name string) error {
 }
 
 // listIGInstances lists all instances of provided instance group name in all zones.
-func (m *manager) listIGInstances(name string) (sets.String, error) {
-	nodeNames := sets.NewString()
+// The return format will be a map from node name to zone.
+func (m *manager) listIGInstances(name string) (map[string]string, error) {
```
Comment (swetharepakula) [inline pkg/instancegroups/manager.go:191]: can you return the string set here?

File: pkg/instancegroups/manager.go [hunk 3]
```diff
@@ -297,10 +298,27 @@ func (m *manager) add(groupName string, names []string) error {
 }
 
 // Remove removes the given instances from the appropriately zoned Instance Group.
-func (m *manager) remove(groupName string, names []string) error {
+func (m *manager) remove(groupName string, names []string, nameZoneMap map[string]string) error {
```
[the remark this answers is not in the document]
Comment (songrx1997) [reply pkg/instancegroups/manager.go]: Done

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**